spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/config/EurekaConfigServerBootstrapper.java

```diff
@@ -45,52 +43,39 @@ public void initialize(BootstrapRegistry registry) {
-	}
-
-	private static Boolean getDiscoveryEnabled(Binder binder) {
-		return binder.bind(ConfigClientProperties.CONFIG_DISCOVERY_ENABLED, Boolean.class).orElse(false)
-				&& binder.bind("eureka.client.enabled", Boolean.class).orElse(true)
-				&& binder.bind("spring.cloud.discovery.enabled", Boolean.class).orElse(true);
-	}
-
-	final static class EurekaFunction implements ConfigServerInstanceProvider.Function {
-
-		private final BootstrapContext context;
-
-		static EurekaFunction create(BootstrapContext context) {
-			return new EurekaFunction(context);
-		}
-
-		private EurekaFunction(BootstrapContext context) {
-			this.context = context;
-		}
-
-		@Override
-		public List<ServiceInstance> apply(String serviceId, Binder binder, BindHandler bindHandler, Log log) {
-			if (binder == null || !getDiscoveryEnabled(binder)) {
-				return Collections.emptyList();
+		registry.registerIfAbsent(EurekaClientConfigBean.class, context -> {
+			if (!getDiscoveryEnabled(context)) {
+				return null;
 			}
-
-			EurekaClientConfigBean config = binder.bind(EurekaClientConfigBean.PREFIX, EurekaClientConfigBean.class)
-					.orElseGet(EurekaClientConfigBean::new);
+			PropertyResolver propertyResolver = getPropertyResolver(context);
+			return propertyResolver.resolveConfigurationProperties(EurekaClientConfigBean.PREFIX,
+					EurekaClientConfigBean.class, EurekaClientConfigBean::new);
+		});
+
+		registry.registerIfAbsent(ConfigServerInstanceProvider.Function.class, context -> {
+			if (!getDiscoveryEnabled(context)) {
+				return (id) -> Collections.emptyList();
+			}
+			EurekaClientConfigBean config = context.get(EurekaClientConfigBean.class);
 			EurekaHttpClient httpClient = new RestTemplateTransportClientFactory(
 					context.getOrElse(TlsProperties.class, null),
 					context.getOrElse(EurekaClientHttpRequestFactorySupplier.class,
-							new DefaultEurekaClientHttpRequestFactorySupplier()))
-									.newClient(HostnameBasedUrlRandomizer.randomEndpoint(config, binder));
-			return new EurekaConfigServerInstanceProvider(httpClient, config).getInstances(serviceId);
-		}
+							new DefaultEurekaClientHttpRequestFactorySupplier())).newClient(
+									HostnameBasedUrlRandomizer.randomEndpoint(config, getPropertyResolver(context)));
+			return new EurekaConfigServerInstanceProvider(httpClient, config)::getInstances;
+		});
+	}
 
-		@Override
-		public List<ServiceInstance> apply(String serviceId) {
-			// This should never be called now but is here for backward
-			// compatibility
-			return apply(serviceId, null, null, null);
-		}
+	private static PropertyResolver getPropertyResolver(BootstrapContext context) {
+		return context.getOrElseSupply(PropertyResolver.class,
+				() -> new PropertyResolver(context.get(Binder.class), context.getOrElse(BindHandler.class, null)));
+	}
 
+	private static Boolean getDiscoveryEnabled(BootstrapContext bootstrapContext) {
```

Comment:
Though this does not affect us at the moment, I'd suggest to make every bootstrapper's `isEnabled` function `public` ([similar to the Zookeeper implementation](https://github.com/spring-cloud/spring-cloud-zookeeper/pull/328#discussion_r1458959524)) and give them a consistent name. People who try to customize these will thank you 😊

Reply:
I can make it public, but I am not going to change the name to avoid more breaking changes

Reply:
@ryanjbaxter Would it really be a breaking change to rename a method that was `private`, i.e. nothing could have used it outside the class?

Reply:
@ryanjbaxter Would it really be a breaking change to rename a method that was `private`, i.e. nothing could have used it outside the class until now? I'd think that right now is the ideal point in time to decide what the best name for these `isEnabled`-type methods would be, and to rename all of them to the chosen name, right when they are made `public`.
(However, I understand that this is not really technically necessary for any reason, it'd just be a nice touch to have a consistent name for the same functionality.)